jwt/api_jwk.py

```diff
@@ -52,6 +52,7 @@ def __init__(self, jwk_data: JWKDict, algorithm: str | None = None) -> None:
         if not has_crypto and algorithm in requires_cryptography:
             raise PyJWKError(f"{algorithm} requires 'cryptography' to be installed.")
 
+        self.algorithm = algorithm
```

Comment:
Imo having both `self.algorithm` and `self.Algorithm` may be too similar

Reply:
Understandable.  Any suggestions?  `self.algorithm_name`?  `self.alg`?  (To match the JWK field)

Reply:
`self.algorithm_name` seems alright, actually I think `self.Algorithm` is the one being wrong, and `self.algorithm_class` or something similar might be better. Anyway better not breaking any public attribute

Reply:
Will do.

Reply:
I'm +1 on @Viicos's suggestion.